packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -522,13 +522,82 @@ export function attach(
     typeof setSuspenseHandler === 'function' &&
     typeof scheduleUpdate === 'function';
 
+  type PendingErrorOrWarning = {|
+    args: Array<any>,
+    fiber: Fiber,
+    type: 'error' | 'warn',
+  |};
+
+  let flushErrorOrWarningUpdatesTimoutID: TimeoutID | null = null;
+  const pendingErrorOrWarnings: Array<PendingErrorOrWarning> = [];
+
+  function onErrorOrWarning(
+    fiber: Fiber,
+    type: 'error' | 'warn',
+    args: Array<any>,
+  ): void {
+    // There are a few places that errors might be logged:
+    // 1. During render (either for initial mount or an update)
+    // 2. During commit effects (both active and passive)
+    // 3. During unmounts (or commit effect cleanup functions)
+    //
+    // Initial render presents a special challenge-
+    // since neither backend nor frontend Store know about a Fiber until it has mounted (and committed).
+    // In this case, we need to hold onto errors until the subsequent commit hook is called.
+    //
+    // Passive effects also present a special challenge-
+    // since the commit hook is called before passive effects, are run,
+    // meaning that we might not want to wait until the subsequent commit hook to notify the frontend.
+    //
+    // For this reason, warnings/errors are sent to the frontend periodically (on a timer).
+    // When processing errors, any Fiber that is not still registered is assumed to be unmounted.
+    pendingErrorOrWarnings.push({
+      fiber,
+      type,
+      // React pushes the component stack to the args later
+      // These are not interesting to the view in React devtools where we already have access to the component stack.
+      // TODO (inline errors) Should this actually be fixed when appending the component stack by not mutating the args?
```

Comment:
We don't need the component stack since we have the actual Fiber. That's all we need, (and a more accurate pointer anyway), to place the error in the tree.

Reply:
Agreed, which is why I copied the arguments to `console.(error|warn)`. My question was if we shouldn't need to copy here and instead stop mutating in https://github.com/facebook/react/blob/1dafc3b6fb076fe8376c4c21b5cba40b8b6b5b8d/packages%2Freact-devtools-shared%2Fsrc%2Fbackend%2Fconsole.js#L180

Reply:
I don't feel strongly about this.

Reply:
Let's stick with copying it here. I don't feel comfortable changing existing code.